source/application/TrustedModule.cpp

```diff
@@ -0,0 +1,92 @@
+﻿/*
+ * Copyright (c) 2017- Hourglass Resurrection Team
+ * Hourglass Resurrection is licensed under GPL v2.
+ * Refer to the file COPYING.txt in the project root.
+ */
+
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+
+#include <algorithm>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "Config.h"
+
+#include "TrustedModule.h"
+
+namespace
+{
+    std::map<DWORD, std::vector<std::wstring>> s_trusted_modules;
+
+    UINT CountSharedPrefixLength(const std::wstring& a, const std::wstring& b)
+    {
+        size_t shortest_path = std::min(a.size(), b.size());
+        UINT i;
+        for (i = 0; i < shortest_path && tolower(a[i]) == tolower(b[i]); ++i);

```

Comment:
I'll add an empty `{}` so it's clearer. I have to return the `i`, so it ends up outside the loop. Can definitely make it `size_t`.